src/crystal/hasher.cr

```diff
@@ -45,12 +68,37 @@ struct Crystal::Hasher
   end
 
   def reference(value)
-    @result = @result * 31 + value.object_id.to_u64
-    self
+    permute(value.object_id.to_u64)
   end
 
   def string(value)
-    value.to_slice.hash(self)
+    string(value.to_slice)
+  end
+
+  @[NoInline]
+  def string(value : Bytes)
```

Comment:
it should be named `slice(value : Bytes)`

Reply:
It is for the case, when slice of bytes is a string.
Ok, maybe it could be `def bytes(value : Bytes)` ?

Reply:
Looks ok

Reply:
Yes, `def bytes(value : Bytes)` or just `def bytes(value)` since other methods don't specify a type either.

Reply:
@ysbaddaden , should I make `Slice(UInt8)#hash` to use `hasher.bytes(self)` as you did in
https://github.com/crystal-lang/crystal/pull/5104/commits/7733d3e462005333ec58c542936469d0d27f8231
?